yt_dlp/extractor/generic.py

```diff
@@ -3164,6 +3164,13 @@ def _real_extract(self, url):
         if sportbox_urls:
             return self.playlist_from_matches(sportbox_urls, video_id, video_title, ie=SportBoxIE.ie_key())
 
+        # Look for embedded Spotify player
+        spotify_urls = re.findall(
+            r'<iframe[^>]+src="(https?://open\.spotify.com/embed/[^"]+)"',
+            webpage)
```

Comment:
Move this into `SpotifyIE._extract_urls`

Reply:
I thought about it. There is a bit of a wrinkle in that `SpotifyIE` is the ‘episode’ extractor, and there is a separate ‘show’ extractor `SpotifyShowIE` for a whole podcast series. Meanwhile, embed URLs can refer both to individual episodes and entire shows (I checked). So neither of the two classes is a particularly good destination for this.